src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -156,6 +160,7 @@ export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
       <SearchBar
         {...searchBarProps}
         showDatePicker={![TopNavMenuItemRenderType.OMITTED, false].includes(showDatePicker!)}
+        data$={props.data$}
```

Comment:
Dont pass the whole data observable as a prop. Just pass the query status as a prop. this might be the reason for the loading state issue that you are seeing